paddle/fluid/framework/ir/graph.cc

```diff
@@ -17,10 +17,110 @@ limitations under the License. */
 #include "paddle/fluid/framework/ir/graph.h"
 #include "paddle/fluid/framework/operator.h"
 
+DEFINE_bool(ssa_program, false, "Convert all blocks in program into SSAgraphs");
+
 namespace paddle {
 namespace framework {
 namespace ir {
 
+#if FLAGS_ssa_program
+Graph::Graph(const ProgramDesc &program) : program_(program) {
```

Comment:
It's unclear to me that when you construct a `Graph` from `Program`, you convert all blocks into sub_graphs, then what's the use of `parent_`? I would suggest to put `global_block` to `parent_`(you can rename it as 'global_graph_' to mapping 'global block'), then other `sub_blocks` into `sub_graphs`